src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferencesSearchEngine.cs

```diff
@@ -153,24 +159,14 @@ private async Task ReportGroupsAsync(ImmutableArray<ISymbol> symbols, Cancellati
             }
         }
 
-        private async Task<HashSet<ProjectId>> GetProjectIdsToSearchAsync(
+        private Task<ImmutableArray<Project>> GetProjectToSearchAsync(
             ImmutableArray<ISymbol> symbols, CancellationToken cancellationToken)
         {
             var projects = _documents != null
                 ? _documents.Select(d => d.Project).ToImmutableHashSet()
                 : _solution.Projects.ToImmutableHashSet();
 
-            var result = new HashSet<ProjectId>();
-
-            foreach (var symbol in symbols)
-            {
-                var dependentProjects = await DependentProjectsFinder.GetDependentProjectsAsync(
-                    _solution, symbol, projects, cancellationToken).ConfigureAwait(false);
-                foreach (var project in dependentProjects)
-                    result.Add(project.Id);
-            }
-
-            return result;
+            return DependentProjectsFinder.GetDependentProjectsAsync(_solution, symbols, projects, cancellationToken);
```

Comment:
N calls down to one.  This is especially relevant as we have to get compilations when doing this work because of IVT.  So being able to do that only once is nice.